device_attributes/index.bs

```diff
@@ -0,0 +1,192 @@
+      1. If no [=Annotated Asset ID=] has been set by the administrator, resolve |promise| with {{undefined}} value.
+      1. [=Queue a global task=] on the [=relevant global object=] of [=this=] using the [=managed data task source=] to [=resolve=] promise with the [=Annotated Asset ID=].
+  1. Return |promise|.
+ 
+## {{getAnnotatedLocation()}} method ## {#getAnnotatedLocation-method}
+ 
+  The <dfn method for="NavigatorManagedData">getAnnotatedLocation()</dfn> method steps are:
+  1. Let |promise| be [=a new promise=].
+  1. Run the following steps in parallel:
+      1. If this API is not called by a managed web application with permission to access device attributes, reject |promise| with a {{NotAllowedError}} DOMException.
+      1. If no [=Annotated Location=] has been set by the administrator, resolve |promise| with {{undefined}} value.
+      1. [=Queue a global task=] on the [=relevant global object=] of [=this=] using the [=managed data task source=] to [=resolve=] promise with the [=Annotated Location=].
+  1. Return |promise|.
+ 
+## {{getDirectoryId()}} method ## {#getDirectoryId-method}
+ 
+  The <dfn method for="NavigatorManagedData">getDirectoryId()</dfn> method steps are:
+  1. Let |promise| be [=a new promise=].
+  1. Run the following steps in parallel:
+      1. If this API is not called by a managed web application with permission to access device attributes, reject |promise| with a {{NotAllowedError}} DOMException.
+      1. [=Queue a global task=] on the [=relevant global object=] of [=this=] using the [=managed data task source=] to [=resolve=] promise with the [=Directory ID=].
+  1. Return |promise|.
+ 
+## {{getHostname()}} method ## {#getHostname-method}
+ 
+  The <dfn method for="NavigatorManagedData">getHostname()</dfn> method steps are:
+  1. Let |promise| be [=a new promise=].
+  1. Run the following steps in parallel:
+      1. If this API is not called by a managed web application with permission to access device attributes, reject |promise| with a {{NotAllowedError}} DOMException.
+      1. If no [=Hostname=] has been set by the administrator, resolve |promise| with {{undefined}} value.
+      1. [=Queue a global task=] on the [=relevant global object=] of [=this=] using the [=managed data task source=] to [=resolve=] promise with the [=Hostname=].
+  1. Return |promise|.
+ 
+## {{getSerialNumber()}} method ## {#getSerialNumber-method}
+ 
+  The <dfn method for="NavigatorManagedData">getSerialNumber()</dfn> method steps are:
+  1. Let |promise| be [=a new promise=].
+  1. Run the following steps in parallel:
+      1. If this API is not called by a managed web application with permission to access device attributes, reject |promise| with a {{NotAllowedError}} DOMException.
+      1. [=Queue a global task=] on the [=relevant global object=] of [=this=] using the [=managed data task source=] to [=resolve=] promise with the [=Serial Number=].
```

Comment:
Where does the UA get a device's serial number? Is this a cross-platform concept?

Reply:
Serial numbers are a generally cross-platform concept. I can find examples of how to read the serial number from the platform on macOS and Windows. The primary limitation for User Agents is that on mobile platforms unprivileged applications don't have access to such identifiers.

Reply:
On mobile platforms, would the management system provide the serial number? Is that doable on all management systems, or does this algorithm need a path for systems that can't provide the serial number?